File: app/src/main/java/com/example/testproject/util/TimeUtil.java
/**
 *
 */
package com.example.testproject.util;


import java.text.SimpleDateFormat;

/**
 * Created by mike.li.
 */
public class TimeUtil {
    public final static String FORMAT_YEAR_MONTH_DAY_TIME = "yyyy-MM-dd HH:mm";
    private static SimpleDateFormat sdf;
    static {
        sdf = new SimpleDateFormat(FORMAT_YEAR_MONTH_DAY_TIME);
    }
    public static String longToString(long time, String formatType) {
        if(time==0)return "";
        sdf.applyPattern(formatType);
        return sdf.format(time);
    }
}

File: app/src/main/java/com/example/testproject/util/TimeUtil.kt
/**
 *
 */
package com.example.testproject.util


import java.text.SimpleDateFormat

/**
 * Created by mike.li.
 */
object TimeUtil {
    const val FORMAT_YEAR_MONTH_DAY_TIME = "yyyy-MM-dd HH:mm"
    private var sdf: SimpleDateFormat? = null

    init {
        sdf = SimpleDateFormat(FORMAT_YEAR_MONTH_DAY_TIME)
    }

    fun longToString(time: Long, formatType: String): String {
        if (time == 0L) return ""
        sdf!!.applyPattern(formatType)
        return sdf!!.format(time)
    }
}
